### PR TITLE
Add info box widgets with CSS stylesheets

### PR DESCRIPTION
**Pull request by edan-bainglass**
Based on #624

This PR adds info box widgets to provide in-app information to users

### CODE REVIEW

**Comment by edan-bainglass** [comment]
@superstar54 @unkcpz can you guys have a look at the failed tests? I don't see how my changes could suddenly trigger the following error, which is failing several structure-related tests!

```
DeprecationWarning: dict interface (SpglibDataset['international']) is deprecated.Use attribute interface ({self.__class__.__name__}.{key}) instead
```

**Comment by superstar54** [comment]
I just created an issue on this: https://github.com/aiidalab/aiidalab-widgets-base/issues/625

**Comment by edan-bainglass** [comment]
@superstar54 I see. So we need to update on our end? I take it this is unrelated to my PR. I would rebase on top of an update PR once that's in?

**Comment by superstar54** [comment]
> Regarding an spglib update PR, do we need it? Can we instead pin to <2.5.0?

No idea yet.

**Comment by danielhollas** [comment]
You can just add the DeprecationWarning to the list of ignored warnings in pyproject.toml

(We have configured pytest so that it turns any warnings into errors, maybe we should change it)

**Review by superstar54 — CHANGES_REQUESTED** [review]
Hi @edan-bainglass , thanks for the nice work.

File: aiidalab_widgets_base/infobox.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    def _on_close(self, _=None):
+        """Hide the first time info box and write existing user token to file."""
+        self.message_box.layout.display = "none"
+        self.closing_message.layout.display = "flex"
+        self._write_existing_user_token_to_file()
+
+    def _write_existing_user_token_to_file(self):
+        """Write a token to file marking the user as an existing user."""
+        with open(".app-user-config", "w") as file:
+            file.write("existing-user")
```
[the remark this answers is not in the document]
Comment (edan-bainglass) [reply aiidalab_widgets_base/infobox.py:103]: I agree. This isn't a great approach. Username would be better, or some other uniquely identifying property. This was just a proof of concept. I was hoping someone with more knowledge of AiiDAlab could provide a better solution :)

File: aiidalab_widgets_base/infobox.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+import ipywidgets as ipw
+
+
+class InfoBox(ipw.VBox):
+    """The `InfoBox` component is used to provide additional info regarding a widget or an app."""
+
+    def __init__(self, **kwargs):
+        """`InfoBox` constructor."""
+        custom_css = kwargs.pop("custom-css", "")
```
[the remark this answers is not in the document]
Comment (edan-bainglass) [reply aiidalab_widgets_base/infobox.py:11]: I actually had it like that initially but changed it for what I vaguely recall was a good reason. If I can't remember it in a day or so, I'll revert to this.

[the remark this answers is not in the document]
Comment (danielhollas) [reply aiidalab_widgets_base/infobox.py:11]: Btw: I think the parameter should be called `cuatom_class`. ;-)

[the remark this answers is not in the document]
Comment (edan-bainglass) [reply aiidalab_widgets_base/infobox.py:11]: I redesigned this part a bit. I think it is clearer. Let me know.

File: aiidalab_widgets_base/infobox.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (edan-bainglass) [reply aiidalab_widgets_base/infobox.py:103]: I really don't like how I've implemented it. Just needed a proof of concept. Open to suggestions. A cookie to the user's browser may be a good approach.

[the remark this answers is not in the document]
Comment (edan-bainglass) [reply aiidalab_widgets_base/infobox.py:103]: @giovannipizzi suggested we keep this as a local config file and had some ideas for its structure. I'll let him explain it.

**Comment by edan-bainglass** [comment]
Moving this PR to the QE App for local testing. If all is good and approved for general use, `InfoBox` classes will move back to the widgets base.

**Pull request closed without merging**